truenas_verify/mtree_verify.py

```diff
@@ -1,58 +1,69 @@
 from collections import namedtuple
+from datetime import datetime, UTC
 from hashlib import file_digest
 import itertools
 from multiprocessing import cpu_count, Pool
 from os import lstat
 from stat import S_ISDIR, S_ISREG, S_ISLNK, S_IMODE
+import re
 import sys
+import syslog
 
 
-LOG_PATH = '/var/log/truenas_verify.log'
+LOG_PATH_NAME = '/var/log/audit/truenas_verify'
 MTREE_FILE_PATH = '/conf/rootfs.mtree'
 CHUNK_SIZE = 1000
-MTREE_ENTRY = namedtuple('MtreeEntry', ['fname', 'mode', 'uid', 'gid', 'type', 'link', 'size', 'sha256'])
+MTREE_FIELDS = ['fname', 'mode', 'uid', 'gid', 'type', 'link', 'size', 'sha256']
+MTREE_ENTRY = namedtuple('MtreeEntry', MTREE_FIELDS, defaults=(None,) * len(MTREE_FIELDS))
 
+# We want to initially split the line into 6 parts with the first being the file name.
+# Following the type field entries may contain 'link' or 'size', but not both.
+# The 'link' or 'size' field is the anchor for the pseudo 'extra' field.
+MTREE_ENTRY_FIELDS = [' mode=', ' gid=', ' uid=', ' type=', ' link=', ' size=']
+
+# The 'extra' field is type dependent.
+MTREE_FILE_FIELDS = [' size=', ' sha256digest=']
```

Comment:
I'm going to suggest an efficiency improvement.  Compile each `re` **once** only rather than regenerate on each file.  Something along the lines of:
```
MTREE_ENTRY_RE = re.compile('|'.join(re.escape(field) for field in MTREE_ENTRY_FIELDS)
MTREE_FILE_RE = re.compile('|'.join(re.escape(field) for field in MTREE_FILE_FIELDS)
```
Then you _could_ eliminate the `split_at_fields` function altogether, and use thusly for example:
```
split_entry = MTREE_ENTRY_RE.split(line[1:])
```

...or you could keep the function, and make the 2nd parameter `pattern`,
```
def split_at_fields(line, pattern):
    return pattern.split(line)
```

Called e.g.
```
split_entry = split_at_fields(line[1:], MTREE_ENTRY_RE)

```